### DISCUSSION TITLE
Mud prompt offset after using PgUp/PgDn

### DISCUSSION THREAD

**Discussion by Silvertower** [scandum/tintin#60] [Bugs]
I'm not sure if this is really a bug, but perhaps more unexpected behavior. It is also behavior that began in TinTin++ 2.01.90 and later--2.01.8 and earlier does not present this issue. I think it is likely directly related to the VT100 changes that were made.

The mud I admin/play on prints a prompt without a newline at the end by default, although we do have the option to print a newline for those that want it.

Using tt++ 2.01.8 or earlier, when using PgUp, followed by a PgDn, and then pressing the enter key, you get the following output (expected):

```
< Name 100(100)hp 100(100)mana 100(100)mv > 
< Name 100(100)hp 100(100)mana 100(100)mv > 
```

Using tt++ 2.01.90 or later, when performing the same operations, you get the following output instead (unexpected):

```
< Name 100(100)hp 100(100)mana 100(100)mv > 
                                            < Name 100(100)hp 100(100)mana 100(100)mv > 
```

This appears to be caused by the cursor "remembering" the column that it was on previously when performing the page up/page down operations.

Configuring my prompt on the mud to force a newline after the prompt 'corrects' the behavior, but I prefer typing my input on the same line as the prompt, rather than on a new line--it also allows for more text to fit on the screen due to less extraneous newlines.

I should note that I have only tested this via an SSH session through PuTTY (version 0.75) connected to a Linux host (Ubuntu 20.04), but I think the behavior is related to how tt++ re-draws the line.

I am wondering if there is a way to prevent this from happening (restore the old behavior), or if not, what section of the tt++ code is responsible for re-drawing the prompt/current line so I can investigate the possibility of a personal patch.

Thank you in advance.

**Comment by scandum** [comment]
Thanks, I've added it to my todo list to fix.

**Comment by scandum** [comment]
This should be fixed in the beta. https://mudhalla.net/tintin-beta.tar.gz

**Comment by Silvertower** [comment]
I can confirm that this fixes the reported issue.
